tests/data_layer/test_data_array.py

```diff
@@ -273,3 +274,40 @@ def test_scalar(self):
 
         df_from_da = da.as_df()
         pd.testing.assert_frame_equal(df_from_da, df)
+
+
+class TestMissingData:
+
+    def test_missing_data_raises(self, small_da):
+        """Should create a pandas.DataFrame from a DataArray
+        """
+        da = small_da
+        da.validate_as_full()
+        da.data[1, 1] = numpy.NaN
+
+        with raises(SmifDataMismatchError):
+            da.validate_as_full()
+
+    def test_missing_data_message(self, small_da):
+        """Should create a pandas.DataFrame from a DataArray
```

Comment:
Comment - 'Should check for NaNs and raise error message if any'?

Can we add a test for non-numeric data here? (The other branch of `DataArray.validate_as_full`)